Public/Src/Utilities/Native/IO/Unix/FileUtilities.Unix.cs

```diff
@@ -668,7 +668,7 @@ public void SetFileAccessControl(string path, FileSystemRights fileSystemRights,
                 currentPermissions &= ~permissions;
             }
 
-            result = SetFilePermissionsForFilePath(path, currentPermissions);
+            result = SetFilePermissionsForFilePath(path, currentPermissions, followSymlink: false);

```

Comment:
```suggestion
            // don't follow symlinks because if we do that we may end up modifying source files via symlink outputs
            result = SetFilePermissionsForFilePath(path, currentPermissions, followSymlink: false);
```